src/proxy/hdrs/XPACK.cc

```diff
@@ -501,25 +501,28 @@ XpackDynamicTable::_make_space(uint64_t extra_space_needed)
     return extra_space_needed == 0;
   }
   uint32_t freed = 0;
-  uint32_t tail  = this->_calc_index(this->_entries_tail, 1);
+  uint32_t tail  = this->_entries_tail;
 
   while (extra_space_needed > freed) {
-    if (this->_entries_head < tail) {
-      break;
-    }
+    tail = this->_calc_index(tail, 1); // Move to the next entry
+
     if (this->_entries[tail].ref_count) {
       break;
     }
     freed += this->_entries[tail].name_len + this->_entries[tail].value_len + ADDITIONAL_32_BYTES;
-    tail   = this->_calc_index(tail, 1);
+    if (this->_entries_head == tail) {
```

Comment:
Shouldn't this block be at the beginning of the loop? What if `_make_space` is called where the table is empty?

Reply:
Or we could also append the check to the condition for the `while` statement.

Reply:
There was an empty conditional at the top of the method.  I moved the check into the while and removed the empty check at the top.

Reply:
This however now changes the result code when the table is empty.  Before it would return true if the table was empty and now it will return false.   I have to look to see if this will cause and issue.  It does pass the unit test either way, which is worrisome.

Reply:
> This however now changes the result code when the table is empty. Before it would return true if the table was empty and now it will return false.

Does it? It still returns true if the table is empty AND `extra_space_needed` is 0. It doesn't seem like changed.

Reply:
It looks like changing the result on an empty table is better since that is what we are expecting in `insert_entry()` and `update_maximum()` size we check to see if the table is not empty already.